pylearn2/termination_criteria/tests/test_init.py

```diff
@@ -48,21 +45,40 @@ def test_epochs(epochs_seen, n):
         assert epochs_seen == n, \
             "%d epochs seen and should be %d" % (epochs_seen, n)
 
-    # Tests for N new epochs
-    train_obj = produce_train_obj(new_epochs=True)
+    # Tests for 5 new epochs
+    train_obj = produce_train_obj(new_epochs=True, max_epochs=5)
+    train_obj.main_loop()
+    test_epochs(train_obj.model.monitor.get_epochs_seen(), 5)
+    train_obj = produce_train_obj(new_epochs=True, max_epochs=5,
+                                  model=train_obj.model)
+    train_obj.main_loop()
+    test_epochs(train_obj.model.monitor.get_epochs_seen(), 2 * 5)
+
+    # Tests for 5 max epochs
+    train_obj = produce_train_obj(new_epochs=False, max_epochs=5)
```

Comment:
```new_epochs``` should not make any difference for models trained from scratch. It should be tested that if model is trained again with ```new_epochs=False``` and ```max_epochs=5```, then ```model.monitor.get_epochs_seen() == 5```